pkg/collector/corechecks/cluster/kubernetesapiserver/events_common.go

```diff
@@ -0,0 +1,147 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2017-present Datadog, Inc.
+
+//go:build kubeapiserver
+// +build kubeapiserver
+
+package kubernetesapiserver
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+
+	v1 "k8s.io/api/core/v1"
+
+	"github.com/DataDog/datadog-agent/pkg/metrics"
+	"github.com/DataDog/datadog-agent/pkg/util/kubernetes"
+	"github.com/DataDog/datadog-agent/pkg/util/kubernetes/apiserver"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+var (
+	hostProviderIDMutex sync.Mutex
+	hostProviderIDCache map[string]string
+)
+
+type eventHostInfo struct {
+	hostname   string
+	nodename   string
+	providerID string
+}
+
+// getDDAlertType converts kubernetes event types into datadog alert types
+func getDDAlertType(k8sType string) metrics.EventAlertType {
+	switch k8sType {
+	case v1.EventTypeNormal:
+		return metrics.EventAlertTypeInfo
+	case v1.EventTypeWarning:
+		return metrics.EventAlertTypeWarning
+	default:
+		log.Debugf("Unknown event type '%s'", k8sType)
+		return metrics.EventAlertTypeInfo
+	}
+}
+
+func getInvolvedObjectTags(involvedObject v1.ObjectReference) []string {
+	tags := []string{
+		fmt.Sprintf("kubernetes_kind:%s", involvedObject.Kind),
```

Comment:
usually we use `kube_` prefix and not `kubernetes_` 
If `kubernetes_kind` is already use, you can keep it, but `kube_kind` can be added.